dev-lang/gnucobol/gnucobol-2.2.recipe

```diff
@@ -0,0 +1,123 @@
+SUMMARY="A free COBOL compiler"
+DESCRIPTION="GnuCOBOL (formerly OpenCOBOL) is a free COBOL compiler."
+HOMEPAGE="https://www.gnu.org/software/gnucobol/"
+COPYRIGHT="2001-2017 Free Software Foundation, Inc
+	Keisuke Nishida
+	Roger While
+	Ron Norman
+	Simon Sobisch
+	Edward Hart
+	Brian Tiffin
+	Sergey Kashyrin
+	Dave Pitts"
+LICENSE="GNU GPL v3
+	GNU LGPL v3
+	GNU FDL v1.3"
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/open-cobol/files/gnu-cobol/$portVersion/gnucobol-$portVersion.tar.gz"
+CHECKSUM_SHA256="925838decd65864b2aa3a4bf1385ce4bc708b942e05e8406945a730d7aab32cb"
+
+ARCHITECTURES="!x86_gcc2 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+libVersion="4.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	gnucobol$secondaryArchSuffix = $portVersion
+	cmd:cobc$commandSuffix = $portVersion
+	cmd:cobcrun$commandSuffix = $portVersion
+	lib:libcob$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libdb$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	gnucobol${secondaryArchSuffix}_devel = $portVersion
+	cmd:cob_config$commandSuffix
+	devel:libcob$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	gnucobol$secondaryArchSuffix == $portVersion base
+	devel:libdb$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libdb$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:awk
+	cmd:bison
+	cmd:find
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+TEST_REQUIRES="
+	cmd:wget
+	cmd:perl
+	"
+
+BUILD()
+{
+	autoreconf -vfi
```

Comment:
If autoreconf is not required in haikoports it shouldn't be done at all.
For a VCS build or "force to use system variant": use autogen.sh, for a tarball: just use the already included versions of the scripts and makefiles.

Reply:
```suggestion
	autoreconf -vfi
```
Some architectures are not known in the generated scripts, hence they require regenerated scripts.

Reply:
Please specify which ones are "not known" before but after which are relevant to the haiku port.
Thank you. Note: while this *may* be relevant to version 2.2 it shouldn't be relevant for 3.1 which uses a newer toolset, is it?

Reply:
We need to re-check this, this was done on work someone did prior to the merge, if autoreconf isn't needed we'll sort it out :) (PS, downloaded the link you gave me over email, just need some time to look into it) 
Thanks for keeping track on this @GitMensch